lib/json-schema/validator.rb

```diff
@@ -474,6 +474,8 @@ def merge_missing_values(source, destination)
 
         if @@json_backend == 'yajl'
           @@serializer = lambda{|o| Yajl::Encoder.encode(o) }
+        elsif @@json_backend == 'json'
+          @@serializer = lambda{|o| JSON.dump(o) }
```

Comment:
I changed the methods that were calling serialize and passing the serialized result to fake_uuid (lines 542 and 549) to use schema.hash.to_s, and this was the result:

#### MRI 2.2.2 (within error range)
```
json-schema from file
                          5.668k (± 7.4%) i/s -    169.162k
json-schema from hash
                          5.210k (± 7.1%) i/s -    155.696k


Comparison:
json-schema from file:     5668.0 i/s
json-schema from hash:     5210.2 i/s - 1.09x slower
```

#### JRuby 1.7.20 (1.21x faster for hash, slightly faster for file)
```
Calculating -------------------------------------
json-schema from file
                       420.000  i/100ms
json-schema from hash
                       830.000  i/100ms
-------------------------------------------------
json-schema from file
                          4.339k (± 3.4%) i/s -    130.200k
json-schema from hash
                          8.414k (± 3.1%) i/s -    252.320k

Comparison:
json-schema from hash:     8414.0 i/s
json-schema from file:     4339.4 i/s - 1.94x slower
```